src/data_loader/data_loaders.py

```diff
@@ -31,8 +31,9 @@ class MnistResultDataLoader(BaseDataLoader):
     Customized MNIST result data loader demo
     Returned data will be in dictionary
     """
-    def __init__(self, result_filename, batch_size, num_workers=1, training=True, name=None):
+    def __init__(self, result_filename, key, batch_size, num_workers=1, training=True, name=None):
         self.result_filename = result_filename
```

Comment:
Should `result_filename` be an attribute of `MnistResultDataLoader`?
Or is it specific for `MnistResultDataset`?

Reply:
If it is, passing `result_filename` and `key` as a dictionary via the config file is a better way for the compliance of the open-closed principle.